app/controllers/api/v2/exchange_rates/period_lists_controller.rb

```diff
@@ -20,12 +20,26 @@ def serialized_period_list
         end
 
         def period_list
-          @period_list ||= ::ExchangeRates::PeriodList.build(year)
+          @period_list ||= ::ExchangeRates::PeriodList.build(year, type)
         end
 
         def year
           (params[:year].presence || ExchangeRateCurrencyRate.max_year).to_i
         end
+
+        def filter_params
+          params.fetch(:filter, {}).permit(:type)
+        end
+
+        def type
+          filter_params[:type]
+        end
```

Comment:
I guess you should remove this

Reply:
yeah I need to refactor and add tests, so don't bother reviewing yet. this is just to unblock you